code/datums/helper_datums/construction_datum.dm

```diff
@@ -62,6 +62,18 @@
 		qdel(holder)
 	return
 
+/datum/construction/proc/spawn_mecha_result()
+	if(result)
+		var/obj/mecha/m = new result(get_turf(holder))
+		var/obj/item/oldcell = locate (/obj/item/stock_parts/cell) in m
+		QDEL_NULL(oldcell)
+		var/obj/item/c = locate(/obj/item/stock_parts/cell) in holder
+		c.forceMove(m)
+		m.cell = c
+		if(ispath(result, /obj/mecha/combat))
+			m.CheckParts(holder.contents)

```

Comment:
Do you actually need this typecheck ? CheckParts is obj/ function isn't it. And shouldn't cell updates be in there.

Reply:
CheckParts calls /obj/mecha/combat/CheckParts(list/parts_list) so only effects combat mechs. However i could move this to /obj/mecha and combine all the checks.